core/src/main/scala/stryker4s/mutants/findmutants/FileCollector.scala

```diff
@@ -56,24 +56,30 @@ class FileCollector(implicit config: Config) extends SourceCollector with Loggin
     */
   private[this] def listFilesBasedOnGit(processRunner: ProcessRunner): Option[Iterable[File]] = {
     processRunner(Command("git ls-files", "--others --exclude-standard --cached"), config.baseDir) match {
-      case Success(files) => Option(files.map(config.baseDir / _).distinct)
+      case Success(files) => Option(files.map(config.baseDir / _).filterNot(isTarget).distinct)
       case Failure(_)     => None
     }
   }
 
   /**
     * List all files from the base directory specified in the Stryker4s basedir config key.
     */
-  private[this] def listAllFiles(): Option[Iterable[File]] = {
-    Option(config.baseDir
+  private[this] def listAllFiles(): Iterable[File] = {
+    config.baseDir
       .listRecursively
-      .filterNot(file => file.pathAsString.contains(s"${pathSeparator}target$pathSeparator"))
-      .toIterable)
+      .filterNot(isTarget)
```

Comment:
Could you move the `filterNot` here and in the other places to `collectFilesToMutate`? I think you could also remove the `stryker4sTmpFiles` as that filters out files that are in the target folder anyway